### PR TITLE
"Extract column" drill thru

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/38964

Screenshot 2024-02-29 at 16 59 48
Screenshot 2024-02-29 at 16 59 51

### CODE REVIEW

File: src/metabase/lib/drill_thru/column_filter.cljc [hunk 1]
```diff
@@ -60,15 +60,18 @@
   (let [next-stage    (->> (lib.util/canonical-stage-index query stage-number)
                            (lib.util/next-stage-number query))
         base          (cond
-                        ;; Not an aggregation: just the input query and stage.
-                        (not= (:lib/source column) :source/aggregations)
+                        ;; Not an aggregation or breakout: just the input query and stage.
+                        (and
```
Comment (ranquild) [inline src/metabase/lib/drill_thru/column_filter.cljc:64]: Not sure but seems like a bug

File: src/metabase/lib/drill_thru/column_extract.cljc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+(defmethod lib.metadata.calculation/display-info-method :drill-thru/column-extract-type
+  [_query _stage-number {:keys [unit]}]
+  {:display-name (lib.temporal-bucket/describe-temporal-unit unit)})
+
+(defmethod lib.drill-thru.common/drill-thru-method :drill-thru/column-extract
+  [_query _stage-number {:keys [query stage-number column]} & [{:keys [unit]}]]
+  (lib.expression/expression
+    query
+    stage-number
+    (lib.temporal-bucket/describe-temporal-unit unit)
```
Comment (ranquild) [inline src/metabase/lib/drill_thru/column_extract.cljc:49]: We need to handle duplicate expression names e.g. with `unique-expression-name` call

File: frontend/src/metabase/querying/utils/drills/column-extract-drill/column-extract-drill.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import type { Drill } from "metabase/visualizations/types/click-actions";
+import type * as Lib from "metabase-lib";
+
+export const columnExtractDrill: Drill<Lib.ColumnExtractDrillThruInfo> = ({
+  query: _query,
+  stageIndex: _stageIndex,
+  drill: _drill,
+}) => {
+  return [];
```
Comment (ranquild) [inline frontend/src/metabase/querying/utils/drills/column-extract-drill/column-extract-drill.ts:9]: Stub for FE changes. An empty array means that no menu items are created

File: src/metabase/lib/drill_thru.cljc [hunk 4]
```diff
@@ -56,6 +57,7 @@
    {:f #'lib.drill-thru.sort/sort-drill,                                         :return-drills-for-dimensions? true}
    {:f #'lib.drill-thru.summarize-column/summarize-column-drill,                 :return-drills-for-dimensions? true}
    {:f #'lib.drill-thru.summarize-column-by-time/summarize-column-by-time-drill, :return-drills-for-dimensions? true}
+   {:f #'lib.drill-thru.column-extract/column-extract-drill,                     :return-drills-for-dimensions? true}
```
Comment (ranquild) [inline src/metabase/lib/drill_thru.cljc:60]: It should be the last item in the list for column-only drills

**Comment by bshepherdson** [comment]
This is a mixed FE/BE PR written mostly by @ranquild with CLJS cleanup and tests from me.

It should be ready to go now, and should complete M1 of this epic #38964 .

**Review by kamilmielnik — APPROVED** [review]
FE changes look good :+1: 

The ["Time style" setting](https://www.metabase.com/docs/latest/configuring-metabase/localization#dates-and-times) and ["Abbreviate names of days and months" setting](https://www.metabase.com/docs/latest/configuring-metabase/localization#dates-and-times) are not respected yet.

**Review by metamben — APPROVED** [review]
LGTM

File: test/metabase/lib/drill_thru/quick_filter_test.cljc [hunk 5]
```diff
@@ -158,7 +158,7 @@
                                             (get-in lib.drill-thru.tu/test-queries ["ORDERS" :aggregated :row "sum"])]]}]}})))
 
 (deftest ^:parallel apply-quick-filter-on-correct-level-test-2
-  (testing "quick-filter not on an aggregation should NOT introduce an new stage"
+  (testing "quick-filter on a breakout should introduce an new stage"
```
[the remark this answers is not in the document]
Comment (ranquild) [reply test/metabase/lib/drill_thru/quick_filter_test.cljc:161]: I think the correct name should be "quick-filter on a breakout should NOT introduce a new stage"

**Pull request merged**